### PR TITLE
[RFC][FSDP] Don't move ignored params / buffers to device

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #108435
* __->__ #108033
* #108032

Since these are ignored by FSDP, don't move them.

Differential Revision: [D48727044](https://our.internmc.facebook.com/intern/diff/D48727044/)

### CODE REVIEW

File: torch/distributed/fsdp/_init_utils.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import torch.distributed.fsdp._traversal_utils as traversal_utils
 import torch.distributed.fsdp.fully_sharded_data_parallel as fsdp_file
 import torch.nn as nn
+from torch import Tensor
```
Comment (awgu) [inline torch/distributed/fsdp/_init_utils.py:27]: nit: If we do this, I feel like we should replace all `torch.Tensor` in the file with `Tensor`, or else it gets fragmented.

Maybe just use `torch.Tensor` on L857?

File: torch/distributed/fsdp/_init_utils.py [hunk 2]
```diff
@@ -884,7 +898,9 @@ def _move_module_to_device(
         # NOTE: This includes moving ignored modules' parameters. If we
```
Comment (awgu) [inline torch/distributed/fsdp/_init_utils.py:898]: We should remove this note.

**Comment by awgu** [comment]
I am on board with the change.

**Review by awgu — APPROVED** [review]

**Comment by awgu** [comment]
Please check lint before landing. Thanks!

File: torch/distributed/fsdp/_init_utils.py [hunk 3]
```diff
@@ -841,6 +851,7 @@ def _get_modules_to_materialize(
 def _move_module_to_device(
     module: nn.Module,
     ignored_params: Set[nn.Parameter],
+    ignored_buffers: Set[torch.Tensor],
     device_from_device_id: Optional[torch.device],
 ) -> None:
     """
```
Comment (ppwwyyxx) [inline torch/distributed/fsdp/_init_utils.py:857]: the docstring below should be updated

**Pull request closed without merging**